Sources/PrimerSDK/Classes/Core/Logging/LogReporter.swift

```diff
@@ -0,0 +1,24 @@
+//
+//  LogReporter.swift
+//  PrimerSDK
+//
+//  Created by Jack Newcombe on 31/10/2023.
+//
+
+import Foundation
+
+private let silentLogger = DefaultLogger(logLevel: .none)
+
+protocol LogReporter {}
+extension LogReporter {
+    
+    var logger: PrimerLogger {
+        return PrimerLogging.shared.logger
+    }
+    
+    static var logger: PrimerLogger {
+        return PrimerLogging.shared.logger
+    }
```

Comment:
What's the difference between the two?

Reply:
The latter is implicitly available to static functions, so instead of calling e.g. `PrimerLogging.shared.logger.info` in a static function body, you can call `logger.info`.

Plus, you're guaranteed the same concrete log reporter in both static and non-static functions

Reply:
Example here: https://github.com/primer-io/primer-sdk-ios/pull/694/commits/d9e055ab9dbf7442ba8683291791c7de4aca7fe6#diff-8e672db59ad5703e14562620d8d5507628aa14e8d0154c166d23ce65e3898565L33